nodeadm/cmd/nodeadm/init/init.go

```diff
@@ -149,6 +149,13 @@ func (c *initCmd) Run(log *zap.Logger, opts *cli.GlobalOptions) error {
 // Various initializations and verifications of the NodeConfig and
 // perform in-place updates when allowed by the user
 func enrichConfig(log *zap.Logger, cfg *api.NodeConfig) error {
+	log.Info("Fetching kubelet version..")
+	kubeletVersion, err := kubelet.GetKubeletVersion()
+	if err != nil {
+		return err
+	}
+	cfg.Status.KubeletVersion = kubeletVersion
+	log.Info("Fetched kubelet version", zap.String("version", kubeletVersion))
```

Comment:
We were calling `kubelet.GetKubeletVersion()` in several places, and they all had access to the `NodeConfig`, so I just cached it in the `NodeConfigStatus`.